biarchtool.py

```diff
@@ -46,28 +46,70 @@ def __init__(self, project):
         self.rdeps = None
         self.whitelist = {
                 'i586': set([
+                    'bzr',
+                    'git',
+                    # _link to baselibs package
+                    'libjpeg62-turbo',
+                    'mercurial',
+                    'subversion',
                     'ovmf'
                     ]),
                 }
+        self.blacklist = {
+                'i586': set([
+                    'belle-sip',
+                    'release-notes-openSUSE',
+                    'openSUSE-EULAs', # translate-toolkit
+                    'skelcd-openSUSE',
+                    'plasma5-workspace',
+                    ]),
+                }
+
+
+    def _has_baselibs_in_files(self, files):
+        for n in files.findall("./entry[@name='baselibs.conf']"):
+            return True
+        return False
+
+    def _is_link(self, files):
+        for n in files.findall("./entry[@name='_link']"):
+            return True
+        return False
 
     def has_baselibs(self, package):
         if package in self._has_baselibs:
             return self._has_baselibs[package]
 
         ret = False
         files = ET.fromstring(self.cached_GET(self.makeurl(['source', self.project, package])))
-        for n in files.findall("./entry[@name='baselibs.conf']"):
+        if self._has_baselibs_in_files(files):
             logger.debug('%s has baselibs', package)
             ret = True
-            return True
+        elif self._is_link(files):
+            files = ET.fromstring(self.cached_GET(self.makeurl(['source', self.project, package], {'expand':'1'})))
```

Comment:
Perhaps replace both of these calls with `osc.core.meta_get_filelist()` returns list of files names. So the above xpath checks could be replaces with `'_link' in files`.